src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs

```diff
@@ -842,15 +838,9 @@ public static FormulaValue RandBetween(IRContext irContext, NumberValue[] args)
             lower = Math.Ceiling(lower);
             upper = Math.Floor(upper);
 
-            lock (_randomizerLock)
-            {
-                if (_random == null)
-                {
-                    _random = new Random();
-                }
+            var random = _defaultRandService;
```

Comment:
`runner.GetService<IRandomService>() ?? _defaultRandService`

Reply:
Right - this just demonstrates for Rand() function. 
Need to apply to these  other functions too.